platform/ios/src/MGLMapView.mm

```diff
@@ -3246,6 +3246,11 @@ - (void)removeStyleClass:(NSString *)styleClass
     }
 
     std::vector<MGLAnnotationTag> annotationTags = [self annotationTagsInRect:rect];
+    
+    if (!annotationTags.size()) {
+        annotationTags = [self shapeAnnotationTagsInRect:rect];
```

Comment:
If a point annotation and a shape annotation are both visible, this method should return both. That’s what I’d expect after reading this public method’s documentation, anyways.